### PR TITLE
net: l2: ieee802154: readability improvements

### PR DESCRIPTION
**Pull request by ghost**
This change set is next in a series working towards #50336 (TSCH protocol implementation).

It improves the readability of certain functions in the IEEE 802.15.4 stack that will be changed in later change sets. This is meant to help reviewers of those later change sets to catch errors by reducing the cognitive load and context required to intuitively understand the code.

As variable name changes also create an intrinsic loss of context by overriding history, the (few) instances where such renames seem adequate has carefully been chosen. The rationale of each rename action is given in the corresponding commit message.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

File: subsys/net/l2/ieee802154/ieee802154.c [hunk 1]
```diff
@@ -251,22 +251,13 @@ static inline void swap_and_set_pkt_ll_addr(struct net_linkaddr *addr, bool comp
 	switch (mode) {
 	case IEEE802154_ADDR_MODE_EXTENDED:
 		addr->len = IEEE802154_EXT_ADDR_LENGTH;
-
-		if (comp) {
-			addr->addr = ll->comp.addr.ext_addr;
-		} else {
-			addr->addr = ll->plain.addr.ext_addr;
-		}
+		addr->addr = has_pan_id ? ll->plain.addr.ext_addr : ll->comp.addr.ext_addr;
```
Comment (ssharks) [inline subsys/net/l2/ieee802154/ieee802154.c:254]: I see here you used a conditional assignment. This is rarely used in Zephyr. I could not find anything on that in the coding guidelines. Maybe stay with the original if else construction.

File: subsys/net/l2/ieee802154/ieee802154.c [hunk 2]
```diff
@@ -251,22 +251,13 @@ static inline void swap_and_set_pkt_ll_addr(struct net_linkaddr *addr, bool comp
...
 	case IEEE802154_ADDR_MODE_SHORT:
 		addr->len = IEEE802154_SHORT_ADDR_LENGTH;
-
-		if (comp) {
-			addr->addr = (uint8_t *)&ll->comp.addr.short_addr;
-		} else {
-			addr->addr = (uint8_t *)&ll->plain.addr.short_addr;
-		}
+		addr->addr = (uint8_t *)(has_pan_id ? &ll->plain.addr.short_addr
+						    : &ll->comp.addr.short_addr);
```
Comment (ssharks) [inline subsys/net/l2/ieee802154/ieee802154.c:260]: Same here

**Review by ssharks — COMMENTED** [review]
Looks good a minor remark

File: subsys/net/l2/ieee802154/ieee802154.c (hunk 1 above)
Comment (rlubos) [reply subsys/net/l2/ieee802154/ieee802154.c:254]: I don't think there's any requirement regarding conditional assignments in the Coding Guidelines, and it's not really that rarely used in Zephyr codebase. IMO it's fine.

**Review by rlubos — APPROVED** [review]

File: subsys/net/l2/ieee802154/ieee802154.c (hunk 1 above)
Comment (fgrandel) [reply subsys/net/l2/ieee802154/ieee802154.c:254]: @ssharks ok for you if I consider this accepted then?

File: subsys/net/l2/ieee802154/ieee802154.c (hunk 2 above)
Comment (fgrandel) [reply subsys/net/l2/ieee802154/ieee802154.c:260]: Duplicate.

**Review by ssharks — APPROVED** [review]
LGTM

**Pull request merged**